### PR TITLE
Prevent duplication of locked shapes, and unlock pasted locked shapes

### PR DESCRIPTION
**Pull request by TodePond**
This PR makes it so that...
- You can't duplicate locked shapes anymore. We were already disabling the button. Now the keyboard shortcut's behavior reflects that too.
- When you paste a locked shape, it now unlocks the pasted shape. Note: Only root level shapes get unlocked. If you paste a  frame that... say... has a locked child inside it, the pasted locked child will stay locked.

### Change type

- [x] `bugfix`
- [ ] `improvement`
- [ ] `feature`
- [ ] `api`
- [ ] `other`

### Release notes

- Prevented duplication of locked shapes.
- Pasting a locked shape onto the canvas now unlocks it.

### CODE REVIEW

File: packages/editor/src/lib/editor/Editor.ts [hunk 1]
```diff
@@ -6211,11 +6211,12 @@ export class Editor extends EventEmitter<TLEventMap> {
 	 */
 	duplicateShapes(shapes: TLShapeId[] | TLShape[], offset?: VecLike): this {
 		this.run(() => {
-			const ids =
+			const _ids =
 				typeof shapes[0] === 'string'
 					? (shapes as TLShapeId[])
 					: (shapes as TLShape[]).map((s) => s.id)
 
+			const ids = this._shouldIgnoreShapeLock ? _ids : this._getUnlockedShapeIds(_ids)
```
Comment (mimecuvalo) [inline packages/editor/src/lib/editor/Editor.ts:6219]: mind adding a test to `duplicate.test.ts`? 🙏

**Review by mimecuvalo — APPROVED** [review]
pre-approving modulo adding a quick test pls!

File: packages/editor/src/lib/editor/Editor.ts (hunk 1 above)
Comment (TodePond) [reply packages/editor/src/lib/editor/Editor.ts:6219]: will do!
